optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -0,0 +1,412 @@
+
+        Args:
+            vae_decoder_path (`Union[str, Path]`):
+                The path of the VAE decoder ONNX model.
+            text_encoder_path (`Union[str, Path]`):
+                The path of the text encoder ONNX model.
+            unet_path (`Union[str, Path]`):
+                The path of the U-NET ONNX model.
+            provider (`str`, defaults to `"CPUExecutionProvider"`):
+                ONNX Runtime provider to use for loading the model. See https://onnxruntime.ai/docs/execution-providers/
+                for possible providers.
+            session_options (`Optional[ort.SessionOptions]`, defaults to `None`):
+                ONNX Runtime session options to use for loading the model. Defaults to `None`.
+            provider_options (`Optional[Dict]`, defaults to `None`):
+                Provider option dictionary corresponding to the provider used. See available options
+                for each provider: https://onnxruntime.ai/docs/api/c/group___global.html . Defaults to `None`.
+        """
+        vae_decoder_session = ORTModel.load_model(vae_decoder_path, provider, session_options, provider_options)
+        text_encoder_session = ORTModel.load_model(text_encoder_path, provider, session_options, provider_options)
+        unet_session = ORTModel.load_model(unet_path, provider, session_options, provider_options)
+
+        return vae_decoder_session, text_encoder_session, unet_session
+
+    def _save_pretrained(
+        self,
+        save_directory: Union[str, Path],
+        vae_decoder_file_name: str = ONNX_WEIGHTS_NAME,
+        text_encoder_file_name: str = ONNX_WEIGHTS_NAME,
+        unet_file_name: str = ONNX_WEIGHTS_NAME,
+        **kwargs,
+    ):
+        """
+        Saves the vae decoder, the text encoder and the unet subcomponents as well as the model configuration to a
+        directory, so that it can be re-loaded using the
+        [`~optimum.onnxruntime.modeling_diffusion.ORT.ORTStableDiffusionPipeline`] class method.
+
+        Args:
+            save_directory (`Union[str, Path`]):
+                The directory where to save the model files.
+            vae_decoder_file_name (`str`, defaults to `optimum.onnxruntime.utils.ONNX_WEIGHTS_NAME`):
+                The VAE decoder model file name. Overwrites the default file name and allows one to save the VAE decoder model
+                with a different name.
+            text_encoder_file_name (`str`, defaults to `optimum.onnxruntime.utils.ONNX_WEIGHTS_NAME`):
+                The text encoder model file name. Overwrites the default file name and allows one to save the text encoder model
+                with a different name.
+            unet_file_name (`str`, defaults to `optimum.onnxruntime.ONNX_WEIGHTS_NAME`):
+                The U-NET model file name. Overwrites the default file name and allows one to save the U-NET model
+                with a different name.
```

Comment:
I would avoid adding a docstring for protected methods, except if this is absolutely needed.
This becomes hard to maintain along time, and usually it does not bring much value.